File: 4356-sum-of-integers-with-maximum-digit-range/sum-of-integers-with-maximum-digit-range.py
class Solution:
    def getrange(self,num:int) -> int :
        minD = float('inf')
        maxD = float('-inf')

        for digit in list(str(num).strip()) :
            minD = min(minD,int(digit))
            maxD = max(maxD,int(digit))
            
        return maxD - minD


    def maxDigitRange(self, nums: list[int]) -> int:
        maxrange = 0
        candidates = []

        for num in nums :
            r = self.getrange(num)
            if r > maxrange :
                maxrange = r 
                candidates = [num]
            elif r==maxrange :
                candidates.append(num)
        return sum(candidates)

File: 4356-sum-of-integers-with-maximum-digit-range/sum-of-integers-with-maximum-digit-range.cpp
class Solution {
public:
    int getRange(int num) {
        int mxD = INT_MIN, mnD = INT_MAX;
        while(num) {
            mxD = max(mxD, num % 10);
            mnD = min(mnD, num % 10);
            num /= 10;
        }
        return mxD - mnD;
    } 
    int maxDigitRange(vector<int>& nums) {
        int maxRange = 0, range;
        vector<int> ranges;
        for(int i = 0; i < nums.size(); i++) {
            range = getRange(nums[i]);
            maxRange = max(maxRange, range);
            ranges.push_back(range);
        }
        int sum = 0;
        for(int i = 0; i < nums.size(); i++) {
            if(maxRange == ranges[i])
                sum += nums[i];
        }
        return sum;
    }
};